tests/integration/test_asr_gradient_step_and_eval.py

```diff
@@ -136,82 +135,24 @@ def test_jasper_training(self):
         )
-        pre_process_params = {
-            'frame_splicing': 1,
-            'features': 64,
-            'window_size': 0.02,
-            'n_fft': 512,
-            'dither': 1e-05,
-            'window': 'hann',
-            'sample_rate': 16000,
-            'normalize': 'per_feature',
-            'window_stride': 0.01,
-        }
-        preprocessing = nemo_asr.AudioToMelSpectrogramPreprocessor(**pre_process_params)
-        jasper_encoder1 = nemo_asr.JasperEncoder(
-            feat_in=jasper_model_definition['AudioToMelSpectrogramPreprocessor']['features'],
-            **jasper_model_definition['JasperEncoder'],
-        )
-        jasper_encoder2 = nemo_asr.JasperEncoder(
-            feat_in=jasper_model_definition['AudioToMelSpectrogramPreprocessor']['features'],
-            **jasper_model_definition['JasperEncoder'],
-        )
-        # mx_max1 = nemo.backends.pytorch.common.SimpleCombiner(mode="max")
-        # mx_max2 = nemo.backends.pytorch.common.SimpleCombiner(mode="max")
-        jasper_decoder1 = nemo_asr.JasperDecoderForCTC(feat_in=1024, num_classes=len(self.labels))
-        jasper_decoder2 = nemo_asr.JasperDecoderForCTC(feat_in=1024, num_classes=len(self.labels))
-
-        ctc_loss = nemo_asr.CTCLossNM(num_classes=len(self.labels))
-
-        # DAG
-        audio_signal, a_sig_length, transcript, transcript_len = dl()
-        processed_signal, p_length = preprocessing(input_signal=audio_signal, length=a_sig_length)
 
-        encoded1, encoded_len1 = jasper_encoder1(audio_signal=processed_signal, length=p_length)
-        encoded2, encoded_len2 = jasper_encoder2(audio_signal=processed_signal, length=p_length)
-        log_probs1 = jasper_decoder1(encoder_output=encoded1)
-        log_probs2 = jasper_decoder2(encoder_output=encoded2)
-        # log_probs = mx_max1(x1=log_probs1, x2=log_probs2)
-        # encoded_len = mx_max2(x1=encoded_len1, x2=encoded_len2)
-        log_probs = log_probs1
-        encoded_len = encoded_len1
-        loss = ctc_loss(
-            log_probs=log_probs, targets=transcript, input_length=encoded_len, target_length=transcript_len,
+        self.nf.train(
+            [loss], callbacks=[callback], optimizer="sgd", optimization_params={"max_steps": 3, "lr": 0.001},
         )
+        self.nf.reset_trainer()
 
-        callback = nemo.core.SimpleLossLoggerCallback(
-            tensors=[loss], print_func=lambda x: logging.info(str(x[0].item()))
-        )
-        # Instantiate an optimizer to perform `train` action
-        optimizer = nemo.backends.pytorch.actions.PtActions()
-        optimizer.train(
-            [loss], callbacks=[callback], optimizer="sgd", optimization_params={"num_epochs": 10, "lr": 0.0003},
-        )
+        # Assert that training loss went down
+        assert loss_list[-1] < loss_list[0]
 
-    @pytest.mark.unclassified
+    @pytest.mark.integration
     def test_quartznet_training(self):
```

Comment:
Hi, is this the test that samples a batch = whole set, what in turn results in lowering the loss at each step?

Can you please add a comment and describe what is happening? - this might be really useful for other people in writing their tests...

Reply:
Yes, all tests in that file do batch gradient descent instead of stochastic gradient descent.

I can add a comment, do you want a doc string for the tests to explain what they're testing? Or do you want me to explain how I coded my test?

Reply:
I won't stop you from adding any additional comments within the code ;) (I actually do it as I treat is a good practice), but doc strings will be sufficient

Reply:
Added docstrings for asr and tts integration tests.